core/http/api/v2/cluster/apis_basic.go

```diff
@@ -440,3 +455,51 @@ func (a *API) GetByName(c *gin.Context) {
 	}
 	response.SuccessWithData(c, resp)
 }
+
+func (a *API) AddFavorite(c *gin.Context) {
+	op := "cluster: add favorite"
+	clusterIDStr := c.Param(common.ParamClusterID)
+	clusterID, err := strconv.ParseUint(clusterIDStr, 10, 0)
+	if err != nil {
+		response.AbortWithRequestError(c, common.InvalidRequestParam, err.Error())
+		return
+	}
+
+	whetherLike := cluster.WhetherLike{IsFavorite: true}
+
+	err = a.clusterCtl.Like(c, uint(clusterID), &whetherLike)
+	if err != nil {
+		if e, ok := perror.Cause(err).(*herrors.HorizonErrNotFound); ok && e.Source == herrors.ClusterInDB {
+			response.AbortWithRPCError(c, rpcerror.NotFoundError.WithErrMsg(err.Error()))
+			return
+		}
+		log.WithFiled(c, "op", op).Errorf("%+v", err)
+		response.AbortWithRPCError(c, rpcerror.InternalError.WithErrMsg(err.Error()))
+		return
+	}
+	response.Success(c)
+}
+
+func (a *API) DeleteFavorite(c *gin.Context) {
+	op := "cluster: delete favorite"
+	clusterIDStr := c.Param(common.ParamClusterID)
+	clusterID, err := strconv.ParseUint(clusterIDStr, 10, 0)
+	if err != nil {
+		response.AbortWithRequestError(c, common.InvalidRequestParam, err.Error())
+		return
+	}
+
+	whetherLike := cluster.WhetherLike{IsFavorite: false}
+
+	err = a.clusterCtl.Like(c, uint(clusterID), &whetherLike)
```

Comment:
Instead of use a struct called WhetherLike, I would like to define a separate func called dislike in this condition. Is that more clear?

Reply:
renamed it with `ToggleLikeStatus`. It's a very simple function with no complicated logic, this would be enough.